# --- SDE-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
#
# Filename: package/.../libcap/0001-opensde-shared-linking.patch
# Copyright (C) 2013 The OpenSDE Project
#
# More information can be found in the files COPYING and README.
#
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- SDE-COPYRIGHT-NOTE-END ---

Fix cross-compiling libcap by explicitely using $CC as linker when linking
the shared library and do not use CFLAGS when linking.

Note:

This patch is kind of OpenSDE specific as we inject 'LD=$arch_target-ld'
into the make options, which is then overwriting the default.

--- a/libcap/Makefile	2013-09-11 09:54:59.268583104 +0200
+++ b/libcap/Makefile	2013-09-11 09:59:57.251523996 +0200
@@ -50,7 +50,7 @@
 	$(RANLIB) $@
 
 $(MINLIBNAME): $(OBJS)
-	$(LD) $(CFLAGS) $(LDFLAGS) -Wl,-soname,$(MAJLIBNAME) -o $@ $^
+	$(CC) -shared $(LDFLAGS) -Wl,-soname,$(MAJLIBNAME) -o $@ $^
 	ln -sf $(MINLIBNAME) $(MAJLIBNAME)
 	ln -sf $(MAJLIBNAME) $(LIBNAME)
 
--- a/pam_cap/Makefile	2013-09-11 11:18:23.108440336 +0200
+++ b/pam_cap/Makefile	2013-09-11 11:18:32.272648916 +0200
@@ -17,7 +17,7 @@
 	install -m 0755 pam_cap.so $(LIBDIR)/security
 
 pam_cap.so: pam_cap.o
-	$(LD) $(LDFLAGS) -o pam_cap.so $< $(LDLIBS)
+	$(CC) -shared $(LDFLAGS) -o pam_cap.so $< $(LDLIBS)
 
 pam_cap.o: pam_cap.c
 	$(CC) $(CFLAGS) $(IPATH) -c $< -o $@
